src/videojs-hls.js

```diff
@@ -377,12 +384,7 @@ videojs.Hls.prototype.cancelSegmentXhr = function() {
  * Abort all outstanding work and cleanup.
  */
 videojs.Hls.prototype.dispose = function() {
-  var player = this.player();
-
-  // remove event handlers
-  player.off('timeupdate', this.fillBuffer);
-  player.off('timeupdate', this.drainBuffer);
-  player.off('waiting', this.drainBuffer);
+  this.stopCheckingBuffer_();
```

Comment:
Definitely not talking about "clever" places to call it but rather if there were other obvious places aside from `dispose` that we would want to stop buffering. I can't really think of any.